### PR TITLE
Feat/vif reduction none

### LINKED ISSUE #3194: Enhancement: Add per-image reduction option to VisualInformationFidelity
### Motivation

The current implementation of `VisualInformationFidelity (VIF)` returns a single scalar for an entire batch of images.  
This behavior makes it difficult to analyze per-image quality when dealing with multiple images.

Many other TorchMetrics metrics, such as `SSIM` and `PSNR`, already support a `reduction='none'` argument, returning per-image values.  
This enhancement would bring VIF in line with the design of these metrics.

### Pitch

- Add a `reduction` argument (`mean` | `none`) to `VisualInformationFidelity`.
- When `reduction='none'`, the metric should return a tensor of shape `(N,)` with per-image VIF values.
- Default behavior remains `mean` for backward compatibility.

## Example Usage
```python
from torchmetrics.image import VisualInformationFidelity
import torch

preds = torch.rand(4, 3, 256, 256)
target = torch.rand(4, 3, 256, 256)

metric = VisualInformationFidelity(reduction='none')
print(metric(preds, target))  # Expected output: tensor of shape (4,)

### PR DESCRIPTION
**Pull request by iamkulbhushansingh**
## What does this PR do?

Enhances the **Visual Information Fidelity (VIF)** metric by adding support for `reduction='none'`.  
This allows users to retrieve per-sample metric values instead of only reduced tensor outputs, bringing the VIF metric in line with other metrics that already support `reduction` modes.

Fixes #3194 

---

<details>
  <summary>Before submitting</summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not required for minor typos or docs)  
- [x] Did you read the [contributor guidelines](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md)?  
- [x] Did you make sure to **update the documentation** (docstrings + examples)?  
- [x] Did you write **unit tests** for the new feature?

</details>

---

## Why is this needed?

- `reduction='none'` enables batch-level debugging, per-sample metric analysis, and flexible custom aggregations.  
- Maintains **API consistency** with other TorchMetrics metrics that already support multiple reduction modes.

---

## Additional Context

- Added **unit tests** covering `reduction='none'` behavior for VIF.  
- Updated **docstring** in python file.

---

## Did you have fun?

That was my first Open Source Commit. 💯  

----
📚 Documentation preview 📚: https://torchmetrics--3195.org.readthedocs.build/en/3195/

### CODE REVIEW

**Comment by iamkulbhushansingh** [comment]
i want to do a clean PR
